src/GregClient/GregClient.cs

```diff
@@ -45,9 +46,22 @@ public ResponseBody ExecuteAndDeserialize(Request m)
             return Execute(m).Deserialize();
         }
 
+        /// <summary>
+        /// Execute the request and deserialize the content.
+        /// </summary>
+        /// <typeparam name="T">The Type of content</typeparam>
+        /// <param name="m">The request.</param>
+        /// <returns>A <see cref="ResponseWithContent{T}"/> or null if there was an error
+        /// in executing the message.</returns>
         public ResponseWithContentBody<T> ExecuteAndDeserializeWithContent<T>(Request m)
         {
             var response = this.ExecuteInternal(m);
+
+            if (response.StatusCode == HttpStatusCode.NotFound)
```

Comment:
This is the only one I care about right now. Greg, in general, doesn't check any error codes anywhere. It just hopes that the reponses contain content and hands that content down to the deserialization.